bulkhead/src/test/java/com/github/tech_salad/resilience/bulkhead/client/RestClientTest.java

```diff
@@ -0,0 +1,195 @@
+    return Stream.of(
+            Arguments.of(this.drinkR4jRestClient, this.saladR4jRestClient, "resilience4j-semaphore"),
+            Arguments.of(this.drinkR4jThreadPoolRestClient, this.saladR4jThreadPoolRestClient, "resilience4j-threadpool"),
+            Arguments.of(this.drinkThreadPoolRestClient, this.saladThreadPoolRestClient, "threadpool")
+    );
+  }
+  @ParameterizedTest(name = "[{index}] => {2}")
+  @MethodSource("provideArguments")
+  void testGetSaladsOnceOK(RestClient<Drink> drinkRestClient, RestClient<Salad> saladRestClient, String testId) {
+    LocalDateTime startTime = LocalDateTime.now();
+
+    // when
+    List<Salad> salads = saladRestClient.get();
+    log.info("{}", salads);
+
+    LocalDateTime endTime = LocalDateTime.now();
+    int duration = (int) Duration.between(startTime, endTime).getSeconds();
+
+    // then
+    assertEquals(2, salads.size());
+
+    assertThat("Bulkhead concurrency = 2, so we should not have executed all in one shot",
+            duration,
+            lessThanOrEqualTo(REST_CALL_DURATION_IN_SECONDS + CALL_OVERHEAD_DURATION));
+  }
+
+  @ParameterizedTest(name = "[{index}] => {2}")
+  @MethodSource("provideArguments")
+  @SneakyThrows
+  void testGetSaladsExceedMaxCapacityPostponesCalls(RestClient<Drink> drinkRestClient, RestClient<Salad> saladRestClient, String testId) {
+    LocalDateTime startTime = LocalDateTime.now();
+
+    // when
+    invokeParallel(saladRestClient, ASYNC_TASKS_COUNT)
+            .stream().forEach(CompletableFuture::join);
+
+    LocalDateTime endTime = LocalDateTime.now();
+    int duration = (int) Duration.between(startTime, endTime).getSeconds();
+
+    assertThat("Bulkhead concurrency = 2, so we should not have exceeded time needed for sequential retrieval",
+            duration,
+            lessThan(ASYNC_TASKS_COUNT * REST_CALL_DURATION_IN_SECONDS));
+    assertThat("Bulkhead concurrency = 2, so we should not have executed all in one shot",
+            duration,
+            greaterThan(REST_CALL_DURATION_IN_SECONDS + CALL_OVERHEAD_DURATION));
+  }
+
+
+  @ParameterizedTest(name = "[{index}] => {2}")
+  @MethodSource("provideArguments")
+  @SneakyThrows
+  void testGetSaladsAndGetDrinksIndependantCalls(RestClient<Drink> drinkRestClient, RestClient<Salad> saladRestClient, String testId) {
+    LocalDateTime startTime = LocalDateTime.now();
+
+    // when
+    Stream.of(
+                    invokeParallel(saladRestClient, ASYNC_TASKS_COUNT / 2),
+                    invokeParallel(drinkRestClient, ASYNC_TASKS_COUNT / 2))
+            .flatMap(List::stream)
+            .forEach(CompletableFuture::join);
+
+    LocalDateTime endTime = LocalDateTime.now();
+    int duration = (int) Duration.between(startTime, endTime).getSeconds();
+
+    assertThat("Bulkhead concurrency = 2, so we should not have exceeded time needed for sequential retrieval",
+            duration,
+            lessThan( 1 + ASYNC_TASKS_COUNT * REST_CALL_DURATION_IN_SECONDS / 4));
```

Comment:
Would it be possible to extract part `1 + ASYNC_TASKS_COUNT * REST_CALL_DURATION_IN_SECONDS / 4` to  variable with some self-describing name?